src/Model/AuditManager.php

```diff
@@ -56,7 +56,10 @@ public function getReader(string $class): AuditReaderInterface
     {
         foreach ($this->readers as $readerId => $classes) {
             if (\in_array($class, $classes, true)) {
-                return $this->container->get($readerId);
+                $reader = $this->container->get($readerId);
+                \assert($reader instanceof AuditReaderInterface);
```

Comment:
Maybe it is better to throw exception if not instance of?

Reply:
Yes indeed.

Looking at this code, it could be weird to throw an exception in the getter, and not throwing one in the setter.
But in the setter, the id passed is not checked.

I know you worked on this class @franmomu for
https://github.com/sonata-project/SonataAdminBundle/issues/6963

The AddAuditReadersCompilerPass is registering the AuditReaderInterface in the service locator.
But If I understand correctly the code, the developer still has to call `setReader`.
Couldn't this be done in the Compiler pass ?

Reply:
Well, in case they are using `sonata-project/doctrine-orm-admin-bundle`, this call is made through [`AddAuditEntityCompilerPass`](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/ff9ca948d8e3f38cc3c02fe5ee49b4151e6a5943/src/DependencyInjection/Compiler/AddAuditEntityCompilerPass.php#L57).

Reply:
@jordisala1991 I added a check in the setter ; what do you think about it ?

Reply:
Not sure when that setter gets called, but if we are checking there on container, why not store directly the class in the array?

Reply:
I'd like to change the code the less possible since I don't know how this work.
I don't know the impact of the call `container->get` (on performance), I'm not sure storing class instance is a good idea, it's the purpose of the container IMHO.

For now, I prefer to do one the following solution
- Check the instance in the setter
- Check the instance in the getter
- Only use assert

And if we can do better, I prefer to let the refacto in another PR.

What do you prefer @sonata-project/contributors ?

Reply:
Since the problem is in the getter, my vote is to add the instance check in the getter. (and it makes things a bit simple, because on the getter we are already retrieving the class from the container, so we can directly do the check, not the same thing for the setter).

Reply:
Done then